orquestaconvert/workflows/base.py

```diff
@@ -251,11 +256,102 @@ def convert_action(self, m_action):
             raise NotImplementedError(("Action '{}' is not supported in orquesta.").
                                       format(m_action))
 
-        return MISTRAL_ACTION_CONVERSION_TABLE.get(m_action, m_action)
+        o_action = MISTRAL_ACTION_CONVERSION_TABLE.get(m_action, m_action)
+
+        kwargs = {'item_vars': self.task_with_item_vars}
+        return MixedExpressionConverter.convert_string(o_action, **kwargs)
+
+    def convert_input(self, input_):
+        kwargs = {'item_vars': self.task_with_item_vars}
+        return MixedExpressionConverter.convert_dict(input_, **kwargs)
+
+    def convert_with_items_expr(self, expression, expr_converter):
```

Comment:
this is a gigantic function and should probably be split up for easier testing and readability